### PR TITLE
LLMS_Section class doc updates

### LINKED ISSUE #1462: Documentation Update
All deprecated methods should include a note in the docblock pointing to a replacement method, where applicable, and if not applicable they should state as much.

+ [x] https://developer.lifterlms.com/reference/classes/llms_section/get_next_available_lesson_order/

### PR DESCRIPTION
**Pull request by thomasplevy**
Updates doc blocks

Closes #1462

### CODE REVIEW

File: includes/models/model.llms.section.php [hunk 1]
```diff
@@ -14,25 +14,44 @@
  * LLMS_Section model class
  *
  * @since 1.0.0
- * @since 3.24.0 Unknown.
  * @since 4.0.0 Remove deprecated class methods.
+ *
+ * @property int    $order         The section's order within its parent course.
+ * @property int    $parent_course The WP_Post ID of the section's parent course.
+ * @property string $title         The title / display name of the section.
```
Comment (eri-trabiccolo) [inline includes/models/model.llms.section.php:21]: not sure about this.. the title is something inherited from the abstract model (it's actually a wp post property).
If we look at the other extending post models looks like we only document the custom (meta) properties...

File: includes/models/model.llms.section.php [hunk 2]
```diff
@@ -117,12 +144,13 @@ public function get_next() {
...
-	 * @param    string $user_id    WP_User ID, if none supplied users current user (if exists)
-	 * @param    bool   $use_cache  when true, uses results from from the wp object cache (if available)
-	 * @return   float
-	 * @since    3.24.0
-	 * @version  3.24.0
+	 * @since 3.24.0
+	 *
+	 * @see LLMS_Student::get_progress()
+	 *
+	 * @param string $user_id   WP_User ID, if none supplied users current user (if exists).
```
Comment (eri-trabiccolo) [inline includes/models/model.llms.section.php]:
```suggestion
	 * @param string $user_id   Optional. WP_User ID, if none supplied uses current user (if exists). Default is empty string.
```

File: includes/models/model.llms.section.php [hunk 3]
```diff
@@ -117,12 +144,13 @@ public function get_next() {
...
+	 * @param bool   $use_cache When true, uses results from from the wp object cache (if available).
```
Comment (eri-trabiccolo) [inline includes/models/model.llms.section.php]:
```suggestion
	 * @param bool   $use_cache Optional. When true, uses results from from the wp object cache (if available). Default is `false`.
```

File: includes/models/model.llms.section.php [hunk 4]
```diff
@@ -160,10 +190,11 @@ public function get_previous() {
...
 	 * Get all lessons in the section
 	 *
-	 * @param    string $return  type of return [ids|posts|lessons]
-	 * @return   int[]|WP_Post[]|LLMS_Lesson[] type depends on value of $return
-	 * @since    3.3.0
-	 * @version  3.24.0
+	 * @since 3.3.0
+	 * @since 3.24.0 Unknown.
+	 *
+	 * @param string $return  type of return [ids|posts|lessons]
```
Comment (eri-trabiccolo) [inline includes/models/model.llms.section.php]:
```suggestion
	 * @param string $return Optional. Type of return [ids|posts|lessons]. Default is `lessons`.
```

File: includes/models/model.llms.section.php [hunk 5]
```diff
@@ -198,10 +229,10 @@ public function get_lessons( $return = 'lessons' ) {
 	/**
 	 * Get sibling sections
 	 *
-	 * @param    string $return  type of return [ids|posts|sections]
-	 * @return   int[]|WP_Post[]|LLMS_Section[] type depends on value of $return
-	 * @since    3.13.0
-	 * @version  3.13.0
+	 * @since 3.13.0
+	 *
+	 * @param string $return Type of return [ids|posts|sections].
```
Comment (eri-trabiccolo) [inline includes/models/model.llms.section.php]:
```suggestion
	 * @param string $return Optional. Type of return [ids|posts|sections]. Default is `sections`.
```

File: includes/models/model.llms.section.php [hunk 6]
```diff
@@ -210,12 +241,14 @@ public function get_siblings( $return = 'sections' ) {
...
-	 * @param    array $arr   data to be serialized
-	 * @return   array
-	 * @since    3.3.0
-	 * @version  3.24.0
+	 * Called before data is sorted and returned by $this->jsonSerialize().
+	 *
+	 * @since 3.3.0
+	 * @since 3.24.0 Unknown.
+	 *
+	 * @param array $arr Data to be serialized
```
Comment (eri-trabiccolo) [inline includes/models/model.llms.section.php]:
```suggestion
	 * @param array $arr Data to be serialized.
```

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]

File: includes/models/model.llms.section.php (hunk 1 above)
Comment (thomasplevy) [reply includes/models/model.llms.section.php:21]: I got yelled at by a developer the other day on Slack for not documenting things well enough so I decided I was going to start documenting the inherited properties too reduce confusion for some

Comment (eri-trabiccolo) [reply includes/models/model.llms.section.php:21]: Ahah I see :)

**Pull request merged**